x/autopilot/keeper/redeem_stake_test.go

```diff
@@ -0,0 +1,253 @@
+			forwardingActive: false,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata(addr1.String(), "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+				Memo:     "",
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     false,
+			expRedeemStake: false,
+		},
+		{ // strd denom
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    strdIbcDenom,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata(addr1.String(), "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+				Memo:     "",
+			},
+			recvDenom:      "ustrd",
+			expSuccess:     false,
+			expRedeemStake: false,
+		},
+		{ // all okay
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata(addr1.String(), "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+				Memo:     "",
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     true,
+			expRedeemStake: true,
+		},
+		{ // all okay with memo liquidstaking since ibc-go v5.1.0
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: addr1.String(),
+				Memo:     getRedeemStakeStakeibcPacketMetadata(addr1.String(), "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     true,
+			expRedeemStake: true,
+		},
+		{ // invalid receiver
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata("xxx", "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k", ""),
+				Memo:     "",
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     false,
+			expRedeemStake: false,
+		},
+		{ // invalid redeem receiver
+			forwardingActive: true,
+			packetData: transfertypes.FungibleTokenPacketData{
+				Denom:    stAtomFullDenomPath,
+				Amount:   "1000000",
+				Sender:   "cosmos16plylpsgxechajltx9yeseqexzdzut9g8vla4k",
+				Receiver: getRedeemStakeStakeibcPacketMetadata(addr1.String(), "xxx", ""),
+				Memo:     "",
+			},
+			recvDenom:      stAtomDenom,
+			expSuccess:     false,
+			expRedeemStake: false,
+		},
+	}
+
+	for i, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %d", i), func() {
+			packet.Data = transfertypes.ModuleCdc.MustMarshalJSON(&tc.packetData)
+
+			suite.SetupTest() // reset
```

Comment:
nit: I don't think this is required - should happen automatically when using `suite.Run`

Reply:
resolved in #1030